storage-xray-udp/src/test/java/zipkin2/storage/xray_udp/UDPMessageEncoderTest.java

```diff
@@ -43,6 +43,30 @@ public void writeJson_server_isSegment() throws Exception {
             "{\"trace_id\":\"1-12345678-90abcdef1234567890abcdef\",\"id\":\"1234567890abcdef\"}");
   }
 
+  @Test
+  public void writeJson_origin_no_default() throws Exception {
+    Span span =
+        serverSpan
+            .toBuilder()
+            .build();
+
+    String json = writeJson(span);
+    assertThat(readString(json, "origin")).isNull();
+  }
+
+  @Test
+  public void writeJson_origin_custom() throws Exception {
+    Span span =
+        serverSpan
+            .toBuilder()
+            .putTag("xray.origin", "AWS::EC2::Instance")
+            .build();
+
+    String json = writeJson(span);
+    assertThat(readString(json, "origin")).isEqualTo("AWS::EC2::Instance");
+    assertThat(readString(json, "annotations.xray_origin")).isEqualTo("AWS::EC2::Instance");
```

Comment:
probably want to drop this tag and prove you did. For example, you can filter out predefined tags. 

ex a loop like this https://github.com/wavefrontHQ/wavefront-spring-boot/pull/55/files#diff-6f5ce475e11d171562ac3a62e4e19d6fR267

(I know the above is brave specific, just mentioning the style of loop. unrelated, but we should do a brave SpanHandler at some point as it is higher fidelity than routing through a sender -> collector etc)